src/jit/flowgraph.cpp

```diff
@@ -3883,15 +3883,38 @@ bool Compiler::fgCreateGCPoll(GCPollType pollType, BasicBlock* block)
         pollType = GCPOLL_CALL;
     }
 
+#ifdef DEBUG
+    // If a statment was supplied it should be contained in the block.
+    if (stmt != nullptr)
+    {
+        bool containsStmt = false;
+        for (Statement* stmtMaybe : block->Statements())
+        {
+            containsStmt = (stmtMaybe == stmt);
+            if (containsStmt)
+            {
+                break;
+            }
+        }
+
+        noway_assert(containsStmt);
```

Comment:
Please use a plain `assert` instead of `noway` under `#ifdef DEBUG`.

Reply:
Will do. For my education, what are the use cases between the two asserts?

Reply:
Will do. For my education, what are the use cases between the two asserts?

Reply:
`assert` fires only in Debug/Checked modes, it is nothing in release.
`noway_assert` behaves as a simple assert in Debug/Checked, but in release it throws an exception, that VM catches and tries to recompile the failed method with `MinOpts`, so the customers don't see a crash.

so `noway_assert` under `#ifdef  DEBUG` is always equal to `assert`.